app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,70 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: [:show, :edit, :update, :destroy]
+  before_action :set_picture
+  before_filter :authenticate_user!, except: [:index, :show]
+  before_action :correct_user, only: [:edit, :update, :destroy]
+
+  # GET /comments
+  def index
+    @comments = Comment.all
+  end
+
+  # GET /comments/1
+  def show
+  end
+
+  # GET /comments/new
+  def new
+    @comment = Comment.new
+  end
+
+  # GET /comments/1/edit
+  def edit
+  end
+
+  # POST /comments
+  def create
+    @comment = @picture.comments.build(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      redirect_to @picture, notice: 'Comment was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # PATCH/PUT /comments/1
+  def update
+    if @comment.update(comment_params)
+      redirect_to @comment, notice: 'Comment was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  # DELETE /comments/1
+  def destroy
+    @comment.destroy
+    redirect_to comments_url, notice: 'Comment was successfully destroyed.'
+  end
+
+  private
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  def set_picture
+    @picture = Picture.find(params[:picture_id])
+  end
+
+  # Only allow a trusted parameter "white list" through.
+  def comment_params
+    params.require(:comment).permit(:body, :picture_id)
+  end
+
+  def correct_user
+    @comment = current_user.comments.find_by(id: params[:id])
+    redirect_to picture_path, notice: 'Not authorized to edit this comment' if @picture.nil?
```

Comment:
Metrics/LineLength: Line is too long. [92/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)